attestation-service/verifier/src/se/ibmse.rs

```diff
@@ -0,0 +1,302 @@
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+extern crate lazy_static;
+use anyhow::{anyhow, bail, Result};
+use log::{debug, info, warn};
+use openssl::{
+    encrypt::{Decrypter, Encrypter},
+    pkey::{PKey, Private, Public},
+    rsa::{Padding, Rsa},
+};
+use pv::{
+    attest::{
+        AdditionalData, AttestationFlags, AttestationItems, AttestationMeasAlg,
+        AttestationMeasurement, AttestationRequest, AttestationVersion,
+    },
+    misc::{open_file, read_certs, read_file},
+    request::{BootHdrTags, CertVerifier, HkdVerifier, ReqEncrCtx, Request, SymKeyType},
+    uv::ConfigUid,
+};
+use serde::{Deserialize, Serialize};
+use serde_json;
+use serde_with::{base64::Base64, serde_as};
+use std::{fs::File, io, io::Read, sync::Mutex};
+
+lazy_static::lazy_static! {
+    static ref PUB_KEY_FILE_CONTENTS: Mutex<Option<Vec<u8>>> = Mutex::new(None);
+    static ref PRI_KEY_FILE_CONTENTS: Mutex<Option<Vec<u8>>> = Mutex::new(None);
+}
+
+fn get_cached_file_or_read(filename: &str, content_ref: &Mutex<Option<Vec<u8>>>) -> io::Result<Vec<u8>> {
+    let mut guard = content_ref.lock().unwrap();
+    if let Some(contents) = guard.as_ref().cloned() {
+        info!("Reading key_file contents from cache.");
+        Ok(contents)
+    } else {
+        info!("Reading key_file contents from file.");
+        let mut file = File::open(filename)?;
+        let mut contents = vec![];
+        file.read_to_end(&mut contents)?;
+        *guard = Some(contents.clone());
+        Ok(contents)
+    }
+}
+
+fn encrypt_measurement_key(key: &[u8], rsa_public_key: &PKey<Public>) -> Result<Vec<u8>> {
+    info!("encrypt_measurement_key.");
+    let mut encrypter = Encrypter::new(rsa_public_key)?;
+    encrypter.set_rsa_padding(Padding::PKCS1)?;
+
+    let buffer_len = encrypter.encrypt_len(key)?;
+    let mut encrypted_hmac_key = vec![0; buffer_len];
+    let len = encrypter.encrypt(key, &mut encrypted_hmac_key)?;
+    encrypted_hmac_key.truncate(len);
+
+    Ok(encrypted_hmac_key)
+}
+
+fn decrypt_measurement_key(key: &[u8], rsa_private_key: &PKey<Private>) -> Result<Vec<u8>> {
+    info!("decrypt_measurement_key.");
+    let mut decrypter = Decrypter::new(rsa_private_key)?;
+    decrypter.set_rsa_padding(Padding::PKCS1)?;
+
+    let buffer_len = decrypter.decrypt_len(key)?;
+    let mut decrypted_hmac_key = vec![0; buffer_len];
+    let decrypted_len = decrypter.decrypt(key, &mut decrypted_hmac_key)?;
+    decrypted_hmac_key.truncate(decrypted_len);
+
+    Ok(decrypted_hmac_key)
+}
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UserData {
+    #[serde_as(as = "Base64")]
+    image_btph: Vec<u8>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SeAttestationClaims {
```

Comment:
for those structs, do you need to set the memory layout to `#[repr(c)]` maybe? If they originate from c libraries rust/llvm might decide to reorder/pad the fields.